pkg/kubelet/cloudresource/cloud_request_manager_test.go

```diff
@@ -97,3 +80,43 @@ func TestNodeAddressesRequest(t *testing.T) {
 	}
 	t.Errorf("Timeout waiting for %q address to appear", cloud.Addresses[0].Address)
 }
+
+func TestNodeAddressesUsesLastSuccess(t *testing.T) {
+	cloud := &fake.FakeCloud{}
+	manager := NewSyncManager(cloud, "defaultNode", 0).(*cloudResourceSyncManager)
+
+	cloud.Addresses = createNodeInternalIPAddress("10.0.1.12")
```

Comment:
Wrap the below in a for loop with test cases. Each case with `cloudAddress string, cloudErr error, wantAddress string`

Reply:
This test is actually testing that the manager falls back correctly to previous node addresses when it encounters an error from the cloud provider.

Reply:
That's fine, test cases can be order-dependent. A comment would help clarify that.

Reply:
Done.